cobigen/cobigen-core/src/test/java/com/devonfw/cobigen/unittest/config/ConfigurationFinderTest.java

```diff
@@ -83,4 +97,210 @@ public void invalidPathTest() {
+    withEnvironmentVariable(ConfigurationConstants.CONFIG_ENV_HOME, userHome.getAbsolutePath()).execute(() -> {
+      URI adapted = ConfigurationFinder.findTemplatesLocation();
+      assertThat(Paths.get(adapted)).exists();
+      assertThat(Paths.get(adapted).getFileName().toString()).isEqualTo(ConfigurationConstants.TEMPLATE_SETS_FOLDER);
+    });
+  }
+
+  /**
+   * Test of findTemplates wit a existing template-set/downloaded folder, expecting to return the template-set folder
+   */
+  @Test
+  public void findTemplateSetDownloadedTest() throws Exception {
+
+    File userHome = this.tmpFolder.newFolder("user-home");
+    this.tmpFolder.newFolder("user-home", ConfigurationConstants.TEMPLATE_SETS_FOLDER,
+        ConfigurationConstants.DOWNLOADED_FOLDER);
+    withEnvironmentVariable(ConfigurationConstants.CONFIG_ENV_HOME, userHome.getAbsolutePath()).execute(() -> {
+      URI downloaded = ConfigurationFinder.findTemplatesLocation();
+      assertThat(Paths.get(downloaded)).exists();
+      assertThat(Paths.get(downloaded).getFileName().toString()).isEqualTo(ConfigurationConstants.TEMPLATE_SETS_FOLDER);
+    });
+  }
+
+  /**
+   * Test of findTemplates wit a existing templates folder with template jars , expecting to return a jar.
+   */
+  @Test
+  public void findTemplatesJarsTest() throws Exception {
+
+    File userHome = this.tmpFolder.newFolder("user-home");
+    File templateS = this.tmpFolder.newFolder("user-home", ConfigurationConstants.TEMPLATES_FOLDER);
+    Files.createFile(templateS.toPath().resolve("templates-devon4j-1.0-sources.jar"));
+    Files.createFile(templateS.toPath().resolve("templates-devon4j-1.0.jar"));
+    withEnvironmentVariable(ConfigurationConstants.CONFIG_ENV_HOME, userHome.getAbsolutePath()).execute(() -> {
+      URI templates = ConfigurationFinder.findTemplatesLocation();
+      assertThat(Paths.get(templates)).exists();
+      assertThat(Paths.get(templates).getFileName().toString()).containsSequence("templates-devon4j-1.0");
+    });
+  }
+
+  /**
+   * Test of findTemplates wit a existing templates folder without template jars , expecting to return a new created
+   * template-Set folder
+   */
+  @Test
+  public void findTemplatesTest() throws Exception {
+
+    File userHome = this.tmpFolder.newFolder("user-home");
+    this.tmpFolder.newFolder("user-home", ConfigurationConstants.TEMPLATES_FOLDER);
+    withEnvironmentVariable(ConfigurationConstants.CONFIG_ENV_HOME, userHome.getAbsolutePath()).execute(() -> {
+      URI templates = ConfigurationFinder.findTemplatesLocation();
+      assertThat(Paths.get(templates)).exists();
+      assertThat(Paths.get(templates).getFileName().toString()).isEqualTo(ConfigurationConstants.TEMPLATE_SETS_FOLDER);
+    });
+  }
+
+  /**
+   * Test of findTemplates wit a existing templates folder with CobiGen_Templates, expecting to return the
+   * CobiGen_Templates folder
+   */
+  @Test
+  public void findTemplatesCGTest() throws Exception {
```

Comment:
Maybe rename to findTemplatesWillReturnCobiGen_Templates?

Reply:
I think findCobiGen_TemplatesTest is a good name, maybe without the underscore